packages/utils/src/common/random_subset.ts

```diff
@@ -22,8 +22,8 @@ function shuffle<T>(arr: T[]): T[] {
 
   for (let i = 0; i < arr.length; i++) {
     const j = randInt();
-    const tmp = arr[i];
-    arr[i] = arr[j];
+    const tmp = arr[i] as T;
+    arr[i] = arr[j] as T;
```

Comment:
does it not automatically infer the type? `arr` is of the type `T[]` 🤔

Reply:
`noUncheckedIndexedAccess` makes so anything that is accepted by index becomes `T | undefined`